requests/utils.py

```diff
@@ -358,13 +358,12 @@ def get_encoding_from_headers(headers):
 
 def stream_decode_response_unicode(iterator, r):
     """Stream decodes a iterator."""
+    encoding = r.encoding
 
     if r.encoding is None:
-        for item in iterator:
-            yield item
-        return
+        encoding = r.apparent_encoding
 
-    decoder = codecs.getincrementaldecoder(r.encoding)(errors='replace')
+    decoder = codecs.getincrementaldecoder(encoding)(errors='replace')
```

Comment:
I added in a unicode error, not sure how descriptive you'd like it. `stream_decode_response_unicode` doesn't seem to currently have any coverage, but I can add tests for the failures if you think that would be useful.